src/nbl/ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -189,103 +197,31 @@ mat2x3 nbl_glsl_MC_getdPos(in vec3 p)
-	}
-	else
-	{
-		stream.offset = InstData.data[InstanceIndex].back_prefetch_offset;
-		stream.count = InstData.data[InstanceIndex].back_prefetch_count;
-	}
-
-	return stream;
-}
-nbl_glsl_instr_stream_t getNormalPrecompStream(in nbl_glsl_MC_precomputed_t precomp)
-{
-	nbl_glsl_instr_stream_t stream;
-	if (precomp.frontface)
-	{
-		stream.offset = InstData.data[InstanceIndex].front_instr_offset + InstData.data[InstanceIndex].front_rem_pdf_count + InstData.data[InstanceIndex].front_genchoice_count;
-		stream.count = InstData.data[InstanceIndex].front_nprecomp_count;
-	}
-	else
-	{
-		stream.offset = InstData.data[InstanceIndex].back_instr_offset + InstData.data[InstanceIndex].back_rem_pdf_count + InstData.data[InstanceIndex].back_genchoice_count;
-		stream.count = InstData.data[InstanceIndex].back_nprecomp_count;
-	}
-
-	return stream;
-}
-
 #ifndef _NBL_BSDF_COS_EVAL_DEFINED_
 #define _NBL_BSDF_COS_EVAL_DEFINED_
 // Spectrum can be exchanged to a float for monochrome
 #define Spectrum vec3
 //! This is the function that evaluates the BSDF for specific view and observer direction
 // params can be either BSDFIsotropicParams or BSDFAnisotropicParams
-Spectrum nbl_bsdf_cos_eval(in nbl_glsl_MC_precomputed_t precomp, in vec3 L, in nbl_glsl_IsotropicViewSurfaceInteraction inter, in mat2 dUV)
+nbl_glsl_MC_precomputed_t precomp;
+nbl_glsl_MC_oriented_material_t material;
+Spectrum nbl_bsdf_cos_eval(in nbl_glsl_LightSample _sample, in nbl_glsl_AnisotropicViewSurfaceInteraction inter, in mat2 dummy_dUV)
 {
-	nbl_glsl_instr_stream_t eval_instrStream = getEvalStream(precomp);
+	nbl_glsl_MC_instr_stream_t eis = nbl_glsl_MC_oriented_material_t_getEvalStream(material);
 
-	return nbl_glsl_runEvalStream(precomp, eval_instrStream, L);
+	return nbl_glsl_MC_runEvalStream(precomp, eis, _sample.L);
 }
 #endif
 
 #ifndef _NBL_COMPUTE_LIGHTING_DEFINED_
 #define _NBL_COMPUTE_LIGHTING_DEFINED_
-vec3 nbl_computeLighting(inout nbl_glsl_IsotropicViewSurfaceInteraction out_interaction, in mat2 dUV, in nbl_glsl_MC_precomputed_t precomp)
+vec3 nbl_computeLighting(inout nbl_glsl_IsotropicViewSurfaceInteraction out_interaction, in mat2 dummy_dUV)
 {
-	vec3 emissive = nbl_glsl_decodeRGB19E7(InstData.data[InstanceIndex].emissive);
-
 	vec3 campos = nbl_glsl_MC_getCamPos();
-	
-	//nbl_glsl_BSDFIsotropicParams params;
-	//params.L = campos-WorldPos;
-	out_interaction = nbl_glsl_calcSurfaceInteraction(campos, WorldPos, normalize(Normal), mat2x3(dFdx(WorldPos),dFdy(WorldPos)));
+	out_interaction = nbl_glsl_calcSurfaceInteraction(campos,WorldPos,normalize(Normal),mat2x3(dFdx(WorldPos),dFdy(WorldPos)));
 
-	return nbl_bsdf_cos_eval(precomp, precomp.V, out_interaction, dUV)/dot(params.L,params.L) + emissive;
+	nbl_glsl_LightSample _sample = nbl_glsl_createLightSample(precomp.V,1.0,precomp.N);
+	return nbl_glsl_MC_oriented_material_t_getEmissive(material)+nbl_bsdf_cos_eval(_sample,out_interaction,dummy_dUV)/dot(params.L,params.L);
```

Comment:
@Crisspl maybe I fucked up here?

something something normalization of `precomp.V`?

Also I think this shader never compiled (`params.L` is undeclared)